QueryBuilder/QueryDSL.swift

```diff
@@ -244,8 +256,26 @@ public protocol SelectionType {
 
 /// Concretely represents a _Selection_ from the GraphQL Language.
 public enum Selection: ObjectSerializable, InlineFragmentSerializable, SelectionType {
-    case scalar(name: String, alias: String?, arguments: [String : InputValue]?, directives: [Directive]?)
-    case object(name: String, alias: String?, arguments: [String : InputValue]?, directives: [Directive]?, selectionSet: SelectionSet)
+    public enum FieldType {
+        case scalar
+        case object(selectionSet: SelectionSet)
+        
+        public mutating func merge(_ other: FieldType) throws {
+            switch (self, other) {
+            case (.scalar, .scalar): return
+            case (.object(var lselectionSet), .object(let rselectionSet)):
+                try lselectionSet.insert(contentsOf: rselectionSet)
+                self = .object(selectionSet: lselectionSet)
+            default:
```

Comment:
You could make this exhaustive with one more case here instead of default, which probably isn't important because you'll probs never add a new FieldType case, but that's all I can think of for this PR.
```swift
case (.object, .scalar), (.scalar, .object):
    ...
```

Reply:
That's actually a good point and easy to change, thanks 👍